prime-router/src/main/kotlin/azure/SubmissionFunction.kt

```diff
@@ -113,62 +114,60 @@ class SubmissionFunction(
         }
     }
 
-    @FunctionName("getSubmissionHistory")
-    fun getSubmissionHistory(
-        @HttpTrigger(
-            name = "getSubmissionHistory",
-            methods = [HttpMethod.GET],
-            authLevel = AuthorizationLevel.ANONYMOUS,
-            route = "history/{organization}/submissions/{submissionId}"
-        ) request: HttpRequestMessage<String?>,
-        @BindingName("organization") organization: String,
-        @BindingName("submissionId") submissionId: Long,
-    ): HttpResponseMessage {
-        return oktaAuthentication.checkAccess(request, organization, true) {
-            try {
-                val submission = facade.findSubmission(organization, submissionId)
-                if (submission != null) HttpUtilities.okJSONResponse(request, submission)
-                else HttpUtilities.notFoundResponse(request, "Submission $submissionId was not found.")
-            } catch (e: DataAccessException) {
-                logger.error("Unable to fetch history for submission ID $submissionId", e)
-                HttpUtilities.internalErrorResponse(request)
-            }
-        }
-    }
-
     /**
-     * Get the history for a given report ID.
+     * API endpoint to return history of a single report.
+     * The [id] can be a valid UUID or a valid actionId (aka submissionId, to our users)
      */
     @FunctionName("getReportHistory")
     fun getReportHistory(
         @HttpTrigger(
             name = "getReportHistory",
             methods = [HttpMethod.GET],
             authLevel = AuthorizationLevel.ANONYMOUS,
-            route = "history/{organization}/report/{reportId}"
+            route = "waters/report/{id}/history"
         ) request: HttpRequestMessage<String?>,
-        @BindingName("organization") organization: String,
-        @BindingName("reportId") reportId: String, // Use string to be able to detect a format error
+        @BindingName("id") id: String,
     ): HttpResponseMessage {
-        return oktaAuthentication.checkAccess(request, organization, true) {
-            val reportUuid = try {
-                UUID.fromString(reportId)
-            } catch (e: IllegalArgumentException) {
-                // Need to isolate this catch as this type of exception can happen for other reasons
-                logger.debug("Invalid format for report ID.", e)
-                null
+        try {
+            val claims = oktaAuthentication.authenticate(request)
+                ?: return HttpUtilities.unauthorizedResponse(request, OktaAuthentication.authenticationFailure)
```

Comment:
This was a big change.   Other code calls `checkAccess` which does both authentication, but then does authorization by looking for an organizationName ...  no such thing in this endpoint, so that code had to be refactored to cleanly separate authn and authz.

The new checkAccess calls the same authenticate(request) as is used here. ^